test/cql-pytest/test_describe.py

```diff
@@ -7,12 +7,572 @@
+def test_table_quoting(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "a int primary key, \"!@#$%\" int") as tbl:
+        desc = cql.execute(f"DESC TABLE {tbl}").one().create_statement
+
+        assert "\"!@#$%\"" in desc
+        assert " !@#$% " not in desc
+
+        cql.execute(f"ALTER TABLE {tbl} DROP \"!@#$%\"")
+        cql.execute(f"ALTER TABLE {tbl} ADD \"!@#$%\" int")
+
+        desc_alter = cql.execute(f"DESC TABLE {tbl} WITH INTERNALS").one().create_statement
+
+        assert "DROP \"!@#$%\"" in desc_alter
+        assert "ADD \"!@#$%\"" in desc_alter
+
+def test_type_quoting(cql, test_keyspace):
+    cql.execute(f"CREATE TYPE {test_keyspace}.\"@@@\" (a int, \"@^@\" text)")
+    desc = cql.execute(f"DESC TYPE {test_keyspace}.\"@@@\"").one().create_statement
+
+    try:
+        assert f"CREATE TYPE {test_keyspace}.\"@@@\"" in desc
+        assert "\"@^@\" text" in desc
+    finally:
+        cql.execute(f"DROP TYPE {test_keyspace}.\"@@@\"")
+
+def test_function_quoting(scylla_only, cql, test_keyspace):
+    cql.execute(f"""
+        CREATE FUNCTION {test_keyspace}.\"@@@\"(val int)
+        RETURNS NULL ON NULL INPUT
+        RETURNS int
+        LANGUAGE lua
+        AS $$ return val $$
+    """)
+    desc = cql.execute(f"DESC FUNCTION {test_keyspace}.\"@@@\"").one().create_statement
+
+    try:
+        assert f"CREATE FUNCTION {test_keyspace}.\"@@@\""
+    finally:
+        cql.execute(f"DROP FUNCTION {test_keyspace}.\"@@@\"")
+
+def test_aggregate_quoting(scylla_only, cql, test_keyspace):
+    cql.execute(f"""
+        CREATE FUNCTION {test_keyspace}.\"@@@\"(val1 int, val2 int)
+        RETURNS NULL ON NULL INPUT
+        RETURNS int
+        LANGUAGE lua
+        AS $$ return val1 + val2 $$
+    """)
+    cql.execute(f"""
+        CREATE AGGREGATE {test_keyspace}.\"!@#\"(int)
+        SFUNC \"@@@\"
+        STYPE int
+    """)
+    desc = cql.execute(f"DESC AGGREGATE {test_keyspace}.\"!@#\"").one().create_statement
+
+    try:
+        assert f"CREATE AGGREGATE {test_keyspace}.\"!@#\"" in desc
+        assert "SFUNC \"@@@\"" in desc
+    finally:
+        cql.execute(f"DROP AGGREGATE {test_keyspace}.\"!@#\"")
+        cql.execute(f"DROP FUNCTION {test_keyspace}.\"@@@\"")
+
+def test_desc_index(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "a int primary key, b int, c int") as tbl:
+        tbl_name = get_name(tbl)
+        create_idx_b = f"CREATE INDEX ON {tbl}(b)"
+        create_idx_c = f"CREATE INDEX named_index ON {tbl}(c)"
+        create_idx_ab = f"CREATE INDEX ON {tbl}((a), b)"
+
+        cql.execute(create_idx_b)
+        cql.execute(create_idx_c)
+        cql.execute(create_idx_ab)
+
+        # I don't know why, but cql return that it cannot find this index in test_keyspace. But the index
+        # exists and when you print result of tbl's describe, this index will be listed.
+        # b_desc = cql.execute(f"DESC INDEX {test_keyspace}.{tbl_name}_b_idx").one().create_statement
```

Comment:
So isn't this a bug in the DESC INDEX code that needs to be debugged?
Do you want to commit this series (which was witheld for too long already) before fixing this? If so, please consider adding an xfailing test (a test which passes on Cassandra, fails on Scylla) for specifically this bug, so we'll have a reminder of this bug more explicit than an "I don't know why" comment.

Reply:
The problem doesn't exist in rewritten version. But I'll keep in mind to use xfails in the future if needed.